### PR TITLE
New Top Bar

### LINKED ISSUE #7411: New top bar in workspace view
image

New design of a top bar which will be visible for users on workspace view. It contains (from the right):
1. Icons which allows user to go to the welcome screen, dashboard and currently open project (active on workspace)
2. Project name and node breadcrumbs
3. Help chat and user profile / settings related stuff

1 & 3 have to be implemented in react and 2nd is already done in Rust. To meet these requirements we (react team) need to render a fixed div and tell rust width of left and right box (so you can move your parts X pixels to the right).

This task assumes:
1. Remove current back to dashboard icon.
2. Allow us to inject the 1&3 boxes width during the EnsoApp constructing next to e.g loader / engine / startup configuration options. 
3. Ensure that events are being properly propagated between the rust / react layers

@somebody1234 anything you would like to add?

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

Fixes #7411 

So far, this branch removes window control buttons and go-to dashboard (hamburger icon), and adds option for dashboard to set offset of the rest of top bar panels.

### Screencast

[screen-recorder-tue-aug-08-2023-15-36-28.webm](https://github.com/enso-org/enso/assets/4046547/ac5ca74e-6a90-4fb1-acbc-e1b399d19de5)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by farmaazon** [comment]
@PabloBuchu @somebody1234 This is the rust part of the task. I assume you now can just add "react-related" stuff to "root" div (with proper z-index, so they will display over the canvas). I don't merge the branch, as it's now impossible to go back to the dashboard atm. Please continue work here.

**Comment by somebody1234** [comment]
react changes pushed. i'm using just z-index, and keeping it in the regular div instead of `#root`, because React needs to be in control of the whole DOM subtree (and so it works better if the react stuff stays in `#enso-dashboard`)

side note: the cursor draws below the React stuff, it might need to be on its own full-screen div with an equal or higher z-index than the React top bar (which has `z-index: 10`)

also if you're aware of some IDE elements using higher numbers than `z-index: 10` do let me know

File: app/gui/view/src/project.rs [hunk 1]
```diff
@@ -281,20 +280,12 @@ impl Model {
...
-        let y = buttons_y - project_view_top_bar_size.y / 2.0;
-        let project_view_top_bar_origin = Vector2(0.0, y);
+        let y = -project_view_top_bar_size.y;
+        let x = ARGS.groups.window.options.top_bar_offset.value;
+        let project_view_top_bar_origin = Vector2(x as f32, y);
         self.top_bar.set_xy(top_left + project_view_top_bar_origin);
     }
 
-    fn on_close_clicked(&self) {
-        js::close(enso_config::window_app_scope_name);
```
Comment (somebody1234) [inline app/gui/view/src/project.rs:295]: i'd assume this means `mod js {}` below should be removed too?

**Comment by somebody1234** [comment]
Note: the traffic lights (window control buttons) are currently *not implemented* in React.
The Figma design seems to indicate they should exist though.
It shouldn't be too hard to implement, but there are some details that would need to be clarified beforehand:
- Maybe that is the macOS system window control buttons, and all that React needs to do is to add padding to its top bar. I do not have a macOS machine, so I cannot confirm that this is how it is intended to work.
- According to the code, it seems like these custom traffic light buttons should only appear when using Enso from the browser?
- The Figma design seems to have all the buttons colored grey
- It is unknown what action each button should do. The fullscreen action should be able to be copied, but the close button seems like it tries to call an API that seems to have been removed for quite a while.

**Comment by farmaazon** [comment]
> * Maybe that is the macOS system window control buttons, and all that React needs to do is to add padding to its top bar. I do not have a macOS machine, so I cannot confirm that this is how it is intended to work.

Yes, these traffics lights are displayed only on macOS, as we remove window frame on that platform, so they are displayed over the scene. We indeed just pushed the top bar right to make them space. You can read the exact values from removed code of my commits in this branch.

> * According to the code, it seems like these custom traffic light buttons should only appear when using Enso from the browser?

This was meant to the old design of cloud, where it was meant to work in the browser; the red button closed the project, yellow put the page into full-screen mode, and green I don't remember. Those should be removed in this branch anyway.

**Comment by somebody1234** [comment]
The app now detects whether it is on macos electron and pushes the top bar to the right.

I think that's all for the JS/React part - @farmaazon is there anything else I might have missed?

**Comment by farmaazon** [comment]
I did a quick check and there are some issues:
1. The mouse pointer is "under" the React icons. It's barely visible on my end, and I think we should just display a system cursor there (like over the documentation panel).
2. When I log out while having project opened, the React icons disappear, and I'm stuck on project view.

image

**Comment by farmaazon** [comment]
Also, the icons on the right seem to not work and does not display "not implemented" caption (like that on the left does).

**Comment by somebody1234** [comment]
just tried, the icons on the right seem to work fine
note that the chat icon and user menu *are* implemented, which is why they don't display any caption on hover

**Comment by farmaazon** [comment]
I thought "cog" icon for setting didn't work for me, but I will double-check.

**Comment by somebody1234** [comment]
it's possible that you were hovering over the blank space between the icons maybe?

that said, it seems like the system cursor re-appears when it's in specific places, which isn't ideal... i'll look into it

**Comment by somebody1234** [comment]
it seems like it's because IDE is changing `cursor: none` back to `cursor: auto` when the cursor enters the top bar, and when moving back there's a one-frame gap where the cursor is still `: auto`

edit: or not. potential fix incoming soon

**Comment by farmaazon** [comment]
Ok, I get "not implemented" tooltip on the right, probably I was too impatient before.

**Comment by somebody1234** [comment]
investigation results:
- it was partly due to React - fixed the React parts
- it is partly because of `show_system_cursor`. setting `cursor: none` (permanently) on `#root` seems to fix it on the IDE side, but i'm not 100% sure removing `show_system_cursor` is the right way to go here - there might be a reason why it *is* actually needed

**Comment by somebody1234** [comment]
because the user menu is now visible in the project menu, it's now possible to sign out while the IDE is still open, leading to a softlock as there is now no way to close the IDE. this definitely needs to be fixed, but i'll be doing other stuff for the time being

**Comment by somebody1234** [comment]
oh, somehow i missed seeing the quick check you did. will address soon

File: app/gui/view/src/project.rs (hunk 1 above)
Comment (vitvakatu) [reply app/gui/view/src/project.rs:295]: @farmaazon I believe so, we don't need to keep this code.

**Comment by vitvakatu** [comment]
QA Report 🔴 

There is one style issue, and one serious functional regression.

1. All icons are incorrectly shifted to the right on Mac. Adam pointed out in his [comment](https://github.com/enso-org/enso/pull/7488#issuecomment-1665208185) that we are removing the window frame on Mac OS. It is not valid for the recent versions. IIRC, we disabled this feature because of some issues with Electron.
2. The execution environment selector is unclickable.
3. The project breadcrumbs respond when hovered over but are unclickable.
4. The entire vertical space under the top bar in the IDE is unclickable.

Screenshot 2023-08-08 at 13 53 50

**Comment by somebody1234** [comment]
@vitvakatu issues should be addressed, mind re-testing?

**Comment by vitvakatu** [comment]
QA passed 🍏

**Review by vitvakatu — APPROVED** [review]
Accepting Rust changes except for comment: https://github.com/enso-org/enso/pull/7488/files#r1286762582

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/dashboard.tsx [hunk 2]
```diff
@@ -212,95 +202,114 @@ export default function Dashboard(props: DashboardProps) {
         }
     }, [/* should never change */ unsetModal])
 
+    const driveHiddenClass = page === pageSwitcher.Page.drive ? '' : 'hidden'
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/dashboard.tsx:211]: there's nowhere else in the codebase that does this, so not sure this should be refactored

**Review by somebody1234 — APPROVED** [review]
not sure whether it's ok to be reviewing my own code so @PabloBuchu @indiv0 feel free to do a code review as well

**Comment by somebody1234** [comment]
i did make some changes since the QA/reviews, hopefully i did enough testing

**Pull request merged**